tests/flytekit/unit/types/structured_dataset/test_structured_dataset_workflow.py

```diff
@@ -28,7 +29,16 @@
 NUMPY_PATH = FlyteContextManager.current_context().file_access.get_random_local_directory()
 BQ_PATH = "bq://flyte-dataset:flyte.table"
 
+
+@dataclass
+class MyCols:
+    Name: str
+    Age: int
+
+
 my_cols = kwtypes(Name=str, Age=int)
+my_dataclass_cols = kwtypes(MyCols)
+my_dict_cols = kwtypes({"Name": str, "Age": int})
 fields = [("Name", pa.string()), ("Age", pa.int32())]
 arrow_schema = pa.schema(fields)
 pd_df = pd.DataFrame({"Name": ["Tom", "Joseph"], "Age": [20, 22]})
```

Comment:
Perhaps add more **`nested`** dataframes to cover extreme test cases.

Reply:
Could we just add your example to the unit test? we can add a new file (test_structured_dataset_workflow_with_nested_type.py) to `tests/flytekit/unit/types/structured_dataset`

Reply:
done